src/fmt/pl.c

```diff
@@ -355,6 +355,31 @@ int pl_bool(bool *val, const struct pl *pl)
 }
 
 
+/**
+ * Convert an ASCII hex string as a pointer-length object to binary format
+ *
+ * @param pl  Pointer-length object
+ * @param hex Destination binary buffer
+ * @param len Length of binary buffer
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int pl_hex(const struct pl *pl, uint8_t *hex, unsigned int len)
+{
+	unsigned int i;
```

Comment:
using C99 style, this variable can be declared inside the for-loop.

Also please use `size_t`